### ISSUE TITLE
✨ Content-Based Filtering (Plot Description)

### ISSUE THREAD

**Issue by CSFelix** [CSFelix/recommendation-system#4]
# ✔️ Studies

- ✔️ Recommender Systems Code: https://www.kaggle.com/code/ibtesama/getting-started-with-a-movie-recommendation-system

- ✔️ TF-IDF: https://www.youtube.com/watch?v=vZAXpvHhQow

- ✔️ Cosine Similarity (Metric): https://www.youtube.com/watch?v=c3xEDTh7Uc8 e https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.cosine_similarity.html

- ✔️ Linear Kernel (Metric): https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.linear_kernel.html

- ✔️ Kernel Trick for SVMs: https://www.youtube.com/watch?v=Q7vT0--5VII

- ✔️ Stemming and Lemmatization in Python: https://www.datacamp.com/tutorial/stemming-lemmatization-python

- ✔️ Part-of-speech tagging: https://campus.datacamp.com/courses/feature-engineering-for-nlp-in-python/text-preprocessing-pos-tagging-and-ner?ex=8

- ✔️ TF-IDF Vectorizer scikit-learn: https://medium.com/@cmukesh8688/tf-idf-vectorizer-scikit-learn-dbc0244a911a

- ✔️ sklearn.feature_extraction.text.TfidfVectorizer: https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.TfidfVectorizer.html (‘l2’: Sum of squares of vector elements is 1. The cosine similarity between two vectors is their dot product when l2 norm has been applied.)

- ✔️ Basic content-based recommendation system with Python code: https://medium.com/mlearning-ai/basic-content-based-recommendation-system-with-python-code-be920b412067

---

# ✔️ Tasks

- ✔️ Content-Based Filtering
- ✔️ Word Cloud
- ✔️ Cosine Similarity
- ✔️ Bag of Words
- ✔️ Token and N-Grams
- ✔️ Stemming and Lemmatization
- ✔️ Stop Words
- ✔️ ZIPF's Law
- ✔️ Term Frequency - Inverse Document Frequency (TF-IDF)
- ✔️ Hands-on
